x/concentrated-liquidity/pool.go

```diff
@@ -143,25 +165,34 @@ func (p Pool) CalcInAmtGivenOut(ctx sdk.Context, tokenOut sdk.Coin, tokenInDenom
 	if tokenOut.Denom == tokenInDenom {
 		return sdk.Coin{}, fmt.Errorf("tokenOut (%s) cannot be the same as tokenIn (%s)", tokenOut.Denom, tokenInDenom)
 	}
+	if minPrice.GTE(curSqrtPrice.Power(2)) {
+		return sdk.Coin{}, fmt.Errorf("minPrice (%s) must be less than current price (%s)", minPrice, curSqrtPrice.Power(2))
+	}
+	if maxPrice.LTE(curSqrtPrice.Power(2)) {
+		return sdk.Coin{}, fmt.Errorf("maxPrice (%s) must be greater than current price (%s)", maxPrice, curSqrtPrice.Power(2))
+	}
 
-	// TODO: Replace with spot price
-	priceLower := sdk.NewDec(4500)
-	priceCur := sdk.NewDec(5000)
-	priceUpper := sdk.NewDec(5500)
-
-	sqrtPLowerTick, _ := priceLower.ApproxSqrt()
-	sqrtPCurTick, _ := priceCur.ApproxSqrt()
-	sqrtPUpperTick, _ := priceUpper.ApproxSqrt()
+	// sqrtPrice of upper and lower user defined price range
+	sqrtPLowerTick, err := minPrice.ApproxSqrt()
+	if err != nil {
+		return sdk.Coin{}, fmt.Errorf("issue calculating square root of minPrice")
+	}
+	sqrtPCurTick := curSqrtPrice
+	sqrtPUpperTick, err := maxPrice.ApproxSqrt()
+	if err != nil {
+		return sdk.Coin{}, fmt.Errorf("issue calculating square root of maxPrice")
+	}
 
-	// TODO: Roman change out with query to pool to get this info
+	// TODO: How do we remove/generalize this? I am stumped.
```

Comment:
Do these values reflect the current spot prices or liquidity?

Reply:
Current liquidity which should be the spot price